lib/BackgroundJobs/SchedulerJob.php

```diff
@@ -24,28 +27,55 @@ class SchedulerJob extends QueuedJob {
 		'OCA\GroupFolders\Mount\MountProvider'
 	];
 
+	public const LOCAL_HOME_MOUNT_TYPE = 'OC\Files\Mount\LocalHomeMountProvider';
+
 	private LoggerInterface $logger;
 	private IDBConnection $db;
 	private IJobList $jobList;
+	/**
+	 * @var \OC\SystemConfig
+	 */
+	private SystemConfig $systemConfig;
 
-	public function __construct(ITimeFactory $timeFactory, Logger $logger, IDBConnection $db, IJobList $jobList) {
+	public function __construct(ITimeFactory $timeFactory, Logger $logger, IDBConnection $db, IJobList $jobList, SystemConfig $systemConfig) {
 		parent::__construct($timeFactory);
 		$this->logger = $logger;
 		$this->db = $db;
 		$this->jobList = $jobList;
+		$this->systemConfig = $systemConfig;
 	}
 
 	protected function run($argument): void {
 		$qb = $this->db->getQueryBuilder();
-		$qb->select('root_id', 'storage_id')
+		$qb->select('root_id', 'storage_id', 'mount_provider_class')
```

Comment:
I wonder if there's a better way for doing all this through the APIs.
I'm guessing that using the APIs would be convoluted and would require iterating over all mounts and/or users just to find the file ids and maybe worse performance. @icewind1991 fyi

maybe we raise a tech debt ticket to address this later